### ISSUE TITLE
Change metabolite from a subclass to a predicate

### ISSUE THREAD

**Issue by vdancik** [biolink/biolink-model#399]
Change **metabolite** from a subclass of **chemical substance** to a predicate _is_metabolite_of_.

**Comment by vdancik** [comment]
While #303 is about modeling biomarkers, some comments there are applicable to metabolites.

**Comment by dkoslicki** [comment]
This recently came to team Expander Agent's attention, and we would like to note the following:

1. **Practical, empirical point of view to keep metabolites as a subclass**
(Via Expander Agent's SME Jared Roach)
Proteins, drugs, and hormones are all also “chemical substances”.  From a practical, empirical point of view, metabolites and proteins are two very importantly distinct classes, because they are assayed in different ways, and the body of scientific literature about them tends to be distinct. It makes sense to reason on them differently and discretely. I don’t send my blood plasma off for a “chemical substance” assay (unless I am a probation officer). Rather, I send it to Metabolon for metabolomics. I might also separately send it to Olink for proteomics.

2.  **Reasoning concern**
If metabolite moves from a sublcass to a predicate, this makes certain questions much more complicated to answer. For example, instead of a simple "What metabolites does Naproxen connect to?" you will need to ask "What chemical substances does Naproxen connect to? Of those, which are connected to something else by the predicate `is_metabolite_of`?" i.e. instead of the (fast)
```
match (n:chemical_substance{id:"CHEBI:7476"})-[]-(m:metabolite) return *
```
you need the (slow, due to the number of chemical substances out there)
```
match p=(n:chemical_substance{id:"CHEBI:7476"})-[]-(m:chemical_substance)
with m as m, p as p
match (m)-[:is_metabolite_of]-(other)
return p
```
In any workflow language, this would require filtering and make it non-trivial to discuss the (as Jared mentions above) concept of metabolites.

**Proposed solution 1.**
Instead of demoting `metabolite` to a predicate, make it a subclass of `chemical_substance` just like is the case with `drug`.

**Proposed solution 2.**
Get rid of subclasses of `chemical_substance` and take advantage of the fact that `category` can be a list. So, for example via Steve Ramsey, `insulin` the `chemical_substance` can have `category` labels of `protein`, `drug`, `hormone`, etc.

I personally prefer solution 1: more consistent with how other subclasses of chemical substances are represented, much less re-tooling of reasoning algos, and still represents the aforementioned nuance of metabolites being chemical substances, but a special subclass of them.

**Comment by vdancik** [comment]
Implemented in PR #582.
